### DISCUSSION TITLE
Doesn't detect first page view

### DISCUSSION THREAD

**Discussion by qlereboursBS** [plausible/analytics#1185] [Self-Hosted Support] [answered]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
**Describe the bug**
I followed the instruction to run the self-hosted version on my computer (localhost), added website but the first page view isn't detected.

What I did (using [these instructions](https://plausible.io/docs/self-hosting)):
 - Pulled and run the docker container with `BASE_URL=http://localhost`
 - Connected as admin on `localhost:8000` (I had to activate accounts with the SQL command line given)
 - Added a new site with host `localhost` and also tried with `localhost:3000`
 - Copy pasted the snippet on my website `<script async defer data-domain="localhost" src="http://localhost/js/plausible.js"></script>`: Please note that it seems to contain an error, it should be `http://localhost:8000/js/plausible.js` since the [default port](https://plausible.io/docs/self-hosting-configuration#server) is 8000, so I replaced this
 - Refresh my `localhost:3000/fr` page multiple times, and checked that `plausible.js` is successfully fetched in the network tab of my browser
But I still see the message "Waiting for first page view"
I checked in the events table of Clickhouse and it's empty.

**Expected behavior**
It should have detected the page view

**Screenshots (If applicable)**
![image](https://user-images.githubusercontent.com/27200095/125574793-f28b02f9-3fc6-4170-9cf1-7362635cbf2c.png)
![image](https://user-images.githubusercontent.com/27200095/125574916-f024df66-5546-4d4c-a75c-cf77435b31c0.png)
![image](https://user-images.githubusercontent.com/27200095/125574958-fdb17b6b-f7b6-47ab-a232-2c209c532939.png)

**Environment (If applicable):**
 - OS: Linux, Ubuntu 20.04
 - Browser Chrome 90
 - The version of the docker image is latest, it should be [this one](https://hub.docker.com/layers/plausible/analytics/latest/images/sha256-3311e01600b82bfc1cec9d4cfc6a0ebf5ec642216a1dfa0d683d929da54b4da0?context=explore)

**Comment by qlereboursBS** [comment, marked as answer]
It seems that it was only related to localhost environment, because after diving in the code I found a condition that prevents to send events on localhost. It shows a warning message, I didn't see it at this moment but it was probably me.
Closing this, but it would be great to have it in the doc somewhere (I didn't find it while following the steps)

**Reply by nvegater** [reply]
Is it working for you ? I still cant get this running on localhost :(

**Reply by nvegater** [reply]
Ok its working now :D just needed to add the `plausible.local.js`. Apart from that I did exactly the same as you described in this issue.
